### ISSUE TITLE
dictutils.OrderedMultiDict.sorted() drops all but the most recently inserted values for each key

### ISSUE THREAD

**Issue by mosbasik** [mahmoud/boltons#209]
## Summary

Running `sorted()` on an instance of `dictutils.OrderedMultiDict` returns an `OMD` that only contains the most recently inserted values.  This seems like unexpected behavior.

## Example
```python
from boltons import dictutils

omd = dictutils.OrderedMultiDict()
omd.add('a', 'apple'  )
omd.add('a', 'apricot')
omd.add('b', 'banana' )

assert len(omd.items(multi=True)) == 3, 'Expect 3 items before sorting.'
omd = omd.sorted()
assert len(omd.items(multi=True)) == 3, 'Expect 3 items after sorting.'  # fails
```

## Cause

[`dictutils.py`, line 486](https://github.com/mahmoud/boltons/blob/master/boltons/dictutils.py#L486) calls `self.iteritems()` without specifying `multi=True`.

## Workaround

In the meantime, the desired result can be achieved by using the same approach used to implement `sorted()`:

```python
dictutils.OrderedMultiDict(sorted(my_ordered_multi_dict.iteritems(multi=True)))
```

## Environment

python 2.7.16
boltons 18.0.0

**Comment by mahmoud** [comment]
Hi @mosbasik, thanks for the detailed error report! You pretty much solved it for me. I also appreciate you generously leaving out the part where I tested it, but the test was wrong. 

Anyways, I fixed it up real quick, and did a quick scout around for any other `mutli=True` args that may have been dropped (there was only one other, in the `__repr__`). 

Thanks again for the great report, and in future, don't hesitate to issue a PR!
